biogrid/libs/biogrid-simulator/src/lib/bioenergy-source/solar-panel.ts

```diff
@@ -0,0 +1,41 @@
+import { getSunlight } from '@biogrid/weather';
+import {EnergySource} from './bioenergy-source'
+import {
+  Validatable,
+  validate,
+  Power,
+  SunlightIntensity,
+} from '@biogrid/grid-simulator';
+
+export class SolarPanel extends EnergySource {
+  private sizeSqMtr: number;
+
+  constructor(sizeSqMtr: number, efficiency = 0.175, longitude = 0, latitude = 0) {
+    super(efficiency, longitude, latitude);
+    if (!this.validateInputsSolarPanel(sizeSqMtr)) {
```

Comment:
is there a reason for using this solar panel specific validation instead of the base class's validator?

Reply:
it uses both, but checking for the size of the panel is specific to solar panels, not all energy users